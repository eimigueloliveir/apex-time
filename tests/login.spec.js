import { describe, expect, it } from 'vitest'
import { mount} from '@vue/test-utils'
import Login from '../pages/index.vue'
import SignIn from  '../pages/sign-in.vue'


describe('Login', () => {
    it('Pode Acessar Login', () => {
        const wrapper = mount(Login)
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('Pode Acessar Cadastro', () => {
        const wrapper = mount(SignIn)
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('Pode Cadastrar', () => {
        const wrapper = mount(SignIn)
        wrapper.find('#username').setValue('teste')
        wrapper.find('#email').setValue('teste@gmail.com')
        wrapper.find('#password').setValue('123456')
    })

})

