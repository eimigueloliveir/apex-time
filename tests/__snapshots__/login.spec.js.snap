// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > Pode Acessar Cadastro 1`] = `
"<nuxtlayout name=\\"login\\">
  <div>
    <div class=\\"loginBox\\">
      <form>
        <h1>Cadastro</h1>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Nome de Usuario\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,user\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Email\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,envelope\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"password\\" required=\\"\\" placeholder=\\"Senha\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,lock\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><button class=\\"btn btn-primary me-3\\" type=\\"submit\\">Cadastrar</button></div>
        <div class=\\"input-box\\">
          <nuxtlink class=\\"register-link\\" to=\\"/\\">Já Tem Uma Conta?</nuxtlink>
        </div>
      </form>
    </div>
  </div>
</nuxtlayout>"
`;

exports[`Login > Pode Acessar Login 1`] = `
"<nuxtlayout name=\\"login\\">
  <div>
    <div class=\\"loginBox\\">
      <form>
        <h1>Login</h1>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Email\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,user\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"password\\" required=\\"\\" placeholder=\\"Senha\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,lock\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><button class=\\"btn btn-primary me-3\\" type=\\"submit\\">Login</button></div>
        <div class=\\"input-box\\">
          <nuxtlink class=\\"register-link\\" to=\\"/sign-in\\">Não Tem Conta?</nuxtlink>
        </div>
      </form>
    </div>
  </div>
</nuxtlayout>"
`;

exports[`Login > Pode Acessar signup 1`] = `
"<nuxtlayout name=\\"login\\">
  <div>
    <div class=\\"loginBox\\">
      <form>
        <h1>Cadastro</h1>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Nome de Usuario\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,user\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Email\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,envelope\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"password\\" required=\\"\\" placeholder=\\"Senha\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,lock\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><button class=\\"btn btn-primary me-3\\" type=\\"submit\\">Cadastrar</button></div>
        <div class=\\"input-box\\">
          <nuxtlink class=\\"register-link\\" to=\\"/\\">Já Tem Uma Conta?</nuxtlink>
        </div>
      </form>
    </div>
  </div>
</nuxtlayout>"
`;

exports[`Login > Pode Cadastrar 1`] = `
"<nuxtlayout name=\\"login\\">
  <div>
    <div class=\\"loginBox\\">
      <form>
        <h1>Cadastro</h1>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Nome de Usuario\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,user\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Email\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,envelope\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"password\\" required=\\"\\" placeholder=\\"Senha\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,lock\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><button class=\\"btn btn-primary me-3\\" type=\\"submit\\">Cadastrar</button></div>
        <div class=\\"input-box\\">
          <nuxtlink class=\\"register-link\\" to=\\"/\\">Já Tem Uma Conta?</nuxtlink>
        </div>
      </form>
    </div>
  </div>
</nuxtlayout>"
`;

exports[`Login > should render Login 1`] = `
"<nuxtlayout name=\\"login\\">
  <div>
    <div class=\\"loginBox\\">
      <form>
        <h1>Login</h1>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"text\\" required=\\"\\" placeholder=\\"Email\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,user\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><input class=\\"form-control\\" type=\\"password\\" required=\\"\\" placeholder=\\"Senha\\">
          <font-awesome-icon class=\\"icon\\" icon=\\"fas,lock\\"></font-awesome-icon>
        </div>
        <div class=\\"input-box\\"><button class=\\"btn btn-primary me-3\\" type=\\"submit\\">Login</button></div>
        <div class=\\"input-box\\">
          <nuxtlink class=\\"register-link\\" to=\\"/sign-in\\">Não Tem Conta?</nuxtlink>
        </div>
      </form>
    </div>
  </div>
</nuxtlayout>"
`;
